src/main/kotlin/com/hypto/iam/server/service/PasscodeService.kt

```diff
@@ -35,21 +37,26 @@ class PasscodeServiceImpl : KoinComponent, PasscodeService {
     override suspend fun verifyEmail(
         email: String,
         purpose: Purpose,
-        organizationId: String?
+        organizationId: String?,
+        metadata: String?
     ): BaseSuccessResponse {
         if (passcodeRepo.getValidPasscodeCount(email, purpose) >= appConfig.app.passcodeCountLimit) {
             throw PasscodeLimitExceededException(
                 "You can only send ${appConfig.app.passcodeCountLimit} passcodes per day"
             )
         }
         val validUntil = LocalDateTime.now().plusSeconds(appConfig.app.passcodeValiditySeconds)
-        val passcode = passcodeRepo.createPasscode(
-            idGenerator.passcodeId(),
-            email,
-            if (purpose == Purpose.signup) null else organizationId,
-            validUntil,
-            purpose
-        )
+        val passcodeRecord = PasscodesRecord().apply {
+            this.id = idGenerator.passcodeId()
+            this.email = email
+            this.purpose = if (purpose == Purpose.signup) null else organizationId
+            this.validUntil = validUntil
+            this.purpose = purpose.toString()
+            this.createdAt = LocalDateTime.now()
+            // TODO: Store metadata in encrypted form as critical information like password is present during signup
```

Comment:
Can we encrypt as part of this PR itself ?

Reply:
As of now we are using Elliptic Curve cryptography, which can only be used for signing the data and not encrypting it.
Can we do that in a different PR as we need to add and store new type of keys also.

Reply:
Added an AES encryption util to encrypt and decrypt metadata using the masterKey(privateKey) as the secret.